src/ORM/DataObject.php

```diff
@@ -2485,6 +2489,65 @@ public function scaffoldFormFields($_params = null)
         return $fs->getFieldList();
     }
 
+    /**
+     * Scaffold a form field for selecting records of this model type in a has_one relation.
+     */
+    public function scaffoldFormFieldForHasOne(
+        string $fieldName,
+        ?string $fieldTitle,
+        string $relationName,
```

Comment:
What's the different between $fieldName and $relationName? Seems like $relationName isn't even used?  For consistenc with the other new methods perhaps rename $fieldName to $relationName and get rid of the 3rd param?

Reply:
`$fieldName` is the name we expect the field to usually have (has ID appended to it). `$relationName` is the name of the actual `has_one` relation.

Relation name is used for `File` (see associated PR in the assets repo), because `UploadField` is weird and doesn't use the real field name.
Giving both here provides flexibility for field types that are non-standard such as `UploadField`.

Reply:
OK. Maybe add docs blocks for the params on this one just to explain why it's the way it is

Reply:
Added - lemme know if you want the wording changed up.